src/Playwright/Core/Waiter.cs

```diff
@@ -218,6 +230,13 @@ internal async Task<T> WaitForPromiseAsync<T>(Task<T> task, Action dispose = nul
             Dispose();
             throw new TimeoutException(ex.Message + FormatLogRecording(_logs), ex);
         }
+        catch (TaskCanceledException ex)
```

Comment:
Catch `OperationCanceledException` instead or both.
See https://github.com/dotnet/runtime/issues/83406#issue-1623962011

Reply:
Yeah, missed that inheritance hierarchy. Fixed